app/src/main/java/com/guru/recognizetext/di/AppModule.kt

```diff
@@ -0,0 +1,21 @@
+package com.guru.recognizetext.di
+
+import android.content.Context
+import com.google.mlkit.vision.text.TextRecognition
+import com.google.mlkit.vision.text.TextRecognizer
+import com.google.mlkit.vision.text.latin.TextRecognizerOptions
+import com.guru.recognizetext.utils.ImageFileManager
+import dagger.Module
+import dagger.Provides
+import javax.inject.Singleton
+
+@Module
+class AppModule(private val context: Context) {
+    @Provides
+    @Singleton
+    fun provideTextRecognizer(): TextRecognizer =
+        TextRecognition.getClient(TextRecognizerOptions.DEFAULT_OPTIONS)
+
+    @Provides
+    fun provideImageFileManager(): ImageFileManager = ImageFileManager(context)
```

Comment:
Check what differences are there if we do or do not use @Singleton and whether we could use it on the whole Module in order to avoid writing @Singleton again and again.

Reply:
Removed @Singleton